test/provider_os_memory_multiple_numa_nodes.cpp

```diff
@@ -0,0 +1,108 @@
+// Copyright (C) 2024 Intel Corporation
+// Under the Apache License v2.0 with LLVM Exceptions. See LICENSE.TXT.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include "base.hpp"
+
+#include <numa.h>
+#include <numaif.h>
+
+#include <umf/providers/provider_os_memory.h>
+
+static umf_os_memory_provider_params_t UMF_OS_MEMORY_PROVIDER_PARAMS_TEST =
+    umfOsMemoryProviderParamsDefault();
+
+std::vector<int> get_available_numa_nodes_numbers() {
+    if (numa_available() == -1 || numa_all_nodes_ptr == nullptr) {
+        return {-1};
+    }
+
+    std::vector<int> available_numa_nodes_numbers;
+    // Get all available NUMA nodes numbers.
+    for (size_t i = 0; i < (size_t)numa_max_node() + 1; ++i) {
+        if (numa_bitmask_isbitset(numa_all_nodes_ptr, i) == 1) {
+            available_numa_nodes_numbers.emplace_back(i);
+        }
+    }
+
+    return available_numa_nodes_numbers;
+}
+
+std::tuple<int, int> retrieve_numa_node_number(void *addr) {
+    int numa_node;
+    int ret =
+        get_mempolicy(&numa_node, nullptr, 0, addr, MPOL_F_NODE | MPOL_F_ADDR);
+
+    return {ret, numa_node};
+}
+
+struct testNumaNodes : public testing::TestWithParam<int> {
+    void SetUp() override {
+        if (numa_available() == -1 || numa_num_task_nodes() <= 1) {
+            GTEST_SKIP();
+        }
+
+        nodemask = numa_allocate_nodemask();
+        ASSERT_NE(nodemask, nullptr);
+    }
+
+    void
+    initOsProvider(umf_os_memory_provider_params_t os_memory_provider_params) {
+        umf_result_t umf_result;
+        umf_result = umfMemoryProviderCreate(&UMF_OS_MEMORY_PROVIDER_OPS,
+                                             &os_memory_provider_params,
+                                             &os_memory_provider);
+        ASSERT_EQ(umf_result, UMF_RESULT_SUCCESS);
+        ASSERT_NE(os_memory_provider, nullptr);
+    }
+
+    void clear() {
+        umf_result_t umf_result;
+        memset(ptr, 0xFF, alloc_size);
+        umf_result = umfMemoryProviderFree(os_memory_provider, ptr, alloc_size);
+        ASSERT_EQ(umf_result, UMF_RESULT_SUCCESS);
+        umfMemoryProviderDestroy(os_memory_provider);
+
+        numa_bitmask_clearall(nodemask);
+        numa_bitmask_free(nodemask);
+        nodemask = nullptr;
```

Comment:
No need for setting the pointer to `nullptr` after freeing the memory in prior calls.

Reply:
Done.